### ISSUE TITLE
add additional liquidity failed

### ISSUE THREAD

**Issue by bthaile** [AugurProject/turbo#324]
<https://kovan.etherscan.io/tx/0x94c934810399ffbb23c5e75904d26c11d1d207d30cbdde24cd37845c3e4e5748>

pool has already been created, getting this error when 1000 collateral liquidity in pool and trying to add 1000 more

![image](https://user-images.githubusercontent.com/3970376/116288140-d48d7b00-a756-11eb-8b99-e7aa74f7b084.png)

<https://dashboard.tenderly.co/tx/kovan/0xcccfc6f4d8a2bf376126e9e2b1dfa3c84f11b4d74c0934c09528a36f2e2a160d/debugger> 

another example. min amount lp tokens is set to zero.

**Comment by robert-e-davidson3** [comment]
This is a bug in how we add liquidity.

When you join a pool, you add tokens in proportion to those already in the pool. So the prices are unchanged but you end up holding tokens.

Right now we join with a single token, repeatedly. That does affect prices but you end up without a position. This method also has the ERR_MAX_IN_RATIO constraint when a balance is very low. For this error, it looks like the balances are zero.

It looks like we need to always be able to do a simple joinPool since it works even if the pool has no balances. But usually we would want to use the second method and join with all of our tokens.

Unfortunately I don't think we can do both in one call. If one of the balances gets extremely low then joinPool will barely increase it. We would need to do repeated swaps. There could be so many that gas limits are hit.

---

Fixing this will affect the UX so we need to have a discussion about how to deal with it.

**Comment by justinbarry** [comment]
tldr; Use join pool to add liquidity and return undeposited shares.
